src/stim/io/measure_record_reader.cc

```diff
@@ -62,6 +60,32 @@ bool read_uint64(FILE *in, uint64_t &value, int &next) {
     return true;
 }
 
+size_t MeasureRecordReader::read_records_into(simd_bit_table &out, bool major_index_is_shot_index, size_t max_shots) {
+    if (!major_index_is_shot_index) {
+        simd_bit_table buf(out.num_minor_bits_padded(), out.num_major_bits_padded());
+        size_t r = read_records_into(buf, true, max_shots);
+        buf.transpose_into(out);
+        return r;
+    }
+
+    size_t rec = 0;
+    max_shots = std::min(max_shots, out.num_major_bits_padded());
+    while (rec < max_shots) {
+        size_t n = read_bytes({out[rec].u8, out[rec].u8 + out[rec].num_u8_padded()});
+        if (n == 0) {
+            break;
+        }
+        if (!is_end_of_record()) {
+            throw std::invalid_argument("Failed to read data. A shot contained more bits than expected.");
+        }
+        rec++;
+        if (!next_record()) {
+            break;
+        }
```

Comment:
This conditional can probably be folded into the while condition.

Reply:
The current semantics of `next_record` don't allow you to call it at the start of reading, so it doesn't fold nicely into the loop.

If it was split into `start_record()` and `skip_rest_of_record()` then it would be possible.